File: mycollab-services/src/main/java/com/esofthead/mycollab/common/SessionIdGenerator.java
package com.esofthead.mycollab.common;

/**
 * 
 * @author MyCollab Ltd.
 * @since 4.3.0
 * 
 */
public abstract class SessionIdGenerator {
	private static SessionIdGenerator instance;

	abstract public String getSessionIdApp();

	public static void registerSessionIdGenerator(SessionIdGenerator provider) {
		instance = provider;
	}

	public static String getSessionId() {
		return instance.getSessionIdApp();
	}
}

File: mycollab-services/src/main/java/com/mycollab/common/SessionIdGenerator.kt
package com.mycollab.common

/**
 * @author MyCollab Ltd.
 * @since 4.3.0
 */
abstract class SessionIdGenerator {

    abstract val sessionIdApp: String

    companion object {
        private lateinit var instance: SessionIdGenerator

        fun registerSessionIdGenerator(provider: SessionIdGenerator) {
            instance = provider
        }

        val sessionId: String
            get() = instance.sessionIdApp
    }
}
